src/realm/error_codes.h

```diff
@@ -35,89 +35,81 @@ typedef enum realm_error_category {
 
 typedef enum realm_errno {
     RLM_ERR_NONE,
-    RLM_ERR_UNKNOWN,
     RLM_ERR_OTHER_EXCEPTION,
+    RLM_ERR_RUNTIME,
+    RLM_ERR_RANGE_ERROR,
+    RLM_ERR_BROKEN_INVARIANT,
     RLM_ERR_OUT_OF_MEMORY,
+    RLM_ERR_OUT_OF_DISK_SPACE,
+    RLM_ERR_ADDRESS_SPACE_EXHAUSTED,
+    RLM_ERR_MAXIMUM_FILE_SIZE_EXCEEDED,
+    RLM_ERR_INCOMPATIBLE_SESSION,
+    RLM_ERR_INCOMPATIBLE_LOCK_FILE,
+    RLM_ERR_INVALID_QUERY,
+    RLM_ERR_DUPLICATE_PRIMARY_KEY_VALUE,
+    RLM_ERR_UNSUPPORTED_FILE_FORMAT_VERSION,
+    RLM_ERR_MULTIPLE_SYNC_AGENTS,
+    RLM_ERR_OBJECT_ALREADY_EXISTS,
     RLM_ERR_NOT_CLONABLE,
+    RLM_ERR_BAD_CHANGESET,
+    RLM_ERR_SUBSCRIPTION_FAILED,
+    RLM_ERR_FILE_OPERATION_FAILED,
+    RLM_ERR_FILE_PERMISSION_DENIED,
+    RLM_ERR_FILE_NOT_FOUND,
+    RLM_ERR_FILE_ALREADY_EXISTS,
+    RLM_ERR_INVALID_DATABASE,
+    RLM_ERR_DECRYPTION_FAILED,
+    RLM_ERR_INCOMPATIBLE_HISTORIES,
+    RLM_ERR_FILE_FORMAT_UPGRADE_REQUIRED,
 
-    RLM_ERR_WRONG_TRANSACTION_STATE,
-    RLM_ERR_WRONG_THREAD,
-
-    RLM_ERR_INVALIDATED_OBJECT,
-    RLM_ERR_INVALID_PROPERTY,
-    RLM_ERR_MISSING_PROPERTY_VALUE,
-    RLM_ERR_PROPERTY_TYPE_MISMATCH,
-    RLM_ERR_OBJECT_TYPE_MISMATCH,
-    RLM_ERR_MISSING_PRIMARY_KEY,
-    RLM_ERR_UNEXPECTED_PRIMARY_KEY,
-    RLM_ERR_MODIFY_PRIMARY_KEY,
-    RLM_ERR_READ_ONLY_PROPERTY,
-    RLM_ERR_PROPERTY_NOT_NULLABLE,
-    RLM_ERR_INVALID_ARGUMENT,
-    RLM_ERR_INVALID_NAME,
+    RLM_ERR_SYSTEM_ERROR,
 
     RLM_ERR_LOGIC,
-    RLM_ERR_NO_SUCH_TABLE,
-    RLM_ERR_NO_SUCH_OBJECT,
+    REALM_ERR_NOT_SUPPORTED,
```

Comment:
This seems the only enum value that starts with REALM instead of RLM.. Is this wanted?

Reply:
That is an oversight :-)